x/batching/keeper/endblock.go

```diff
@@ -49,21 +52,44 @@ func (k Keeper) EndBlock(ctx sdk.Context) (err error) {
 // It returns a resulting batch, data result tree entries, and validator
 // tree entries in that order.
 func (k Keeper) ConstructBatch(ctx sdk.Context) (types.Batch, [][]byte, [][]byte, error) {
-	// Note current will be "old" for this new batch.
-	oldBatchNum, err := k.GetCurrentBatchNum(ctx)
+	var newBatchNum uint64
+	var latestDataRoot, latestValRoot string
+	latestBatch, err := k.GetLatestBatch(ctx)
 	if err != nil {
-		return types.Batch{}, nil, nil, err
+		if !errors.Is(err, types.ErrBatchingHasNotStarted) {
+			return types.Batch{}, nil, nil, err
+		}
+		newBatchNum = collections.DefaultSequenceStart + 1
+	} else {
+		newBatchNum = latestBatch.BatchNumber + 1
+		latestDataRoot = latestBatch.DataResultRoot
+		latestValRoot = latestBatch.ValidatorRoot
```

Comment:
Maybe it makes sense to sufffix these with `Hex`?

Reply:
that's what I did before seeing this comment !